source/classification1.md

```diff
@@ -1523,39 +1430,25 @@ what the data would look like if the cancer was rare. We will do this by
 picking only 3 observations from the malignant group, and keeping all
 of the benign observations. We choose these 3 observations using the `.head()`
 method, which takes the number of rows to select from the top (`n`).
-The new imbalanced data is shown in {numref}`fig:05-unbalanced`.
-
-```{code-cell} ipython3
-:tags: [remove-cell]
-
-# To better illustrate the problem, let's revisit the scaled breast cancer data, 
-# `cancer`; except now we will remove many of the observations of malignant tumors, simulating
-# what the data would look like if the cancer was rare. We will do this by
-# picking only 3 observations from the malignant group, and keeping all
-# of the benign observations. We choose these 3 observations using the `slice_head`
-# function, which takes two arguments: a data frame-like object,
-# and the number of rows to select from the top (`n`).
-# The new imbalanced data is shown in Figure \@ref(fig:05-unbalanced).
-```
+We use the [`concat`](https://pandas.pydata.org/docs/reference/api/pandas.concat.html) 
+function from `pandas` to glue the two resulting filtered
+data frames back together by passing them together in a sequence.
+The new imbalanced data is shown in {numref}`fig:05-unbalanced`, 
+and we print the counts of the classes using the `value_counts` function.
 
 ```{code-cell} ipython3
-cancer = pd.read_csv("data/wdbc.csv")
 rare_cancer = pd.concat(
-    (cancer.query("Class == 'B'"), cancer.query("Class == 'M'").head(3))
-)
-colors = ["#86bfef", "#efb13f"]
-rare_cancer["Class"] = rare_cancer["Class"].apply(
-    lambda x: "Malignant" if (x == "M") else "Benign"
-)
+    (cancer[cancer["Class"] == 'Benign'],
+     cancer[cancer["Class"] == 'Malignant'].head(3)
+    ))
```

Comment:
```suggestion
    cancer[cancer["Class"] == 'Benign'],
    cancer[cancer["Class"] == 'Malignant'].head(3)
))
```

Nice this new approach is much more intuitive than the one with `apply`! I moved this parent up one line